pom.xml

```diff
@@ -467,7 +467,7 @@
           <dependency>
             <groupId>com.puppycrawl.tools</groupId>
             <artifactId>checkstyle</artifactId>
-            <version>7.8</version>
+            <version>7.9-SNAPSHOT</version>
```

Comment:
It looks like it does work. I can run the checkstyle phase without the dependency installed.
If a dependency is already installed, it still seems to use the current project and not the installed one.
I'm still looking if maven documents this behavior somewhere.